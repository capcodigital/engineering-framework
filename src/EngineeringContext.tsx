import { createContext } from 'react';

const EngineeringContext = createContext({
  name: '',
  setName: (name: string) => {},
});

export default EngineeringContext;
