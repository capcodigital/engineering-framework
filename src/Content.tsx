import React, { useContext, useState, useEffect, FC } from 'react';
import { EngineeringContext } from './EngineeringContext';
import { Collapse, Divider, List, ListItem } from '@material-ui/core';
import SideBar from './SideBar';
import { Tag } from './Tag';
import { titles } from './data/data';
import contentData from './data/content';
import images from './img/';
import styled from 'styled-components/macro';
import BreadcrumbsNav from './BreadcrumbsNav';
import SideBarList from './SideBarList';

type ContentType = {
  title: 'Software Engineer' | 'Quality Engineer';
};

type ContentHeaderProps = {
  level: string;
};

type ContentContainerProps = {
  menuOpen: boolean;
};

const Main = styled.main`
  display: flex;
  height: 80vh;
  @media screen and (max-width: 1000px) {
    display: block;
  }
`;

export const ContentContainer = styled.div<ContentContainerProps>`
  width: 100%;
  background-color: #1f1f1f;
  color: white;
  text-align: left;
  display: ${({ menuOpen }) => menuOpen && 'none'};
  @media screen and (max-width: 1000px) {
    ::-webkit-scrollbar {
      display: none;
    }
  }
`;

export const ContentHeader = styled.div<ContentHeaderProps>`
  width: 100%;
  height: 120px;
  padding: 25px 25px 25px 100px;
  background-size: 1280px;
  background-image: url(${({ level }) => (images as any)[level]});

  .level-title {
    font-size: 16px;
    line-height: 24px;
    .level {
      font-weight: bold;
    }
  }
  .category {
    font-size: 36px;
    font-weight: bold;
  }

  @media screen and (max-width: 1000px) {
    height: 96px;
    padding: 16px 25px 25px 25px;
    .level-title {
      font-size: 14px;
      line-height: 16px;
    }
    .category {
      font-size: 32px;
    }
  }
`;

const ContentDiv = styled.div`
  height: calc(100% - 120px);
  overflow: auto;
  ::-webkit-scrollbar {
    display: none;
  }
  .competency-list {
    padding: 0;
  }
  li {
    display: block;
  }
  li.list-item {
    padding: 24px 100px 24px 100px;
  }
  .title {
    width: 100%;
    font-size: 24px;
    font-weight: bold;
    padding-bottom: 16px;
    span {
      margin-right: 20px;
    }
  }
  .description {
    line-height: 24px;
    font-size: 16px;
  }
  hr {
    margin: 48px 0px 24px 0px;
    background-color: white;
  }
  @media screen and (max-width: 1000px) {
    overflow: unset;

    padding: 0;
    li.list-item {
      padding: 24px 28px 24px 28px;
    }
  }
`;

const Description = styled.p`
  font-size: 16px;
  line-height: 24px;
  li {
    display: list-item;
    margin-bottom: 16px;
  }
`;

const Content: FC<ContentType> = ({ title }) => {
  const [open, setOpen] = useState(false);
  const {
    specialism,
    level,
    category,
    setSpecialism,
    setCategory,
    setCompetency,
  } = useContext(EngineeringContext);

  useEffect(() => {
    setSpecialism(title);
  }, [title, setSpecialism]);

  useEffect(() => {
    if (window.location.hash) {
      const [cat, ...comp] = window.location.hash.split('#')[1].split('-');
      const categoryfromUrl = cat.charAt(0).toUpperCase() + cat.slice(1);
      const competencyFromUrl = comp
        .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
        .join(' ');
      setCategory(categoryfromUrl);
      setCompetency(competencyFromUrl);
    }
  }, []);

  useEffect(() => {
    setTimeout(() => {
      const { hash } = window.location;
      if (hash) {
        const id = hash.replace('#', '');
        const element = document.getElementById(id);
        if (element) {
          element.scrollIntoView({ block: 'start', behavior: 'smooth' });
        }
      }
    }, 100);
  }, []);

  const handleClickOpen = () => {
    setOpen(!open);
  };

  return (
    <Main>
      <SideBar>
        <SideBarList />
      </SideBar>

      <BreadcrumbsNav open={open} onClick={() => handleClickOpen()}>
        <Collapse in={open} timeout='auto' unmountOnExit>
          <SideBarList />
        </Collapse>
      </BreadcrumbsNav>

      <ContentContainer menuOpen={open}>
        <ContentHeader level={level}>
          <div className='level-title'>
            <span className='level'>{level}</span> â€¢{' '}
            {(titles as any)[specialism][level]}
          </div>
          <div className='category'>{category}</div>
        </ContentHeader>
        <ContentDiv>
          <List component='div' className={'competency-list'} data-testid='competency-list'>
            {(contentData as any)[level][category].map(
              (comp: any) =>
                [specialism, 'All'].includes(comp.specialism) && (
                  <ListItem
                    id={`${category.toLowerCase()}-${comp.name
                      .replace(/\s/g, '-')
                      .toLowerCase()}`}
                    className='list-item'
                    key={`content-${comp.name}`}
                    alignItems='flex-start'
                    style={{
                      background:
                        comp.name === 'Framework Criteria' ? '#161616' : '',
                    }}
                  >
                    <div className='title'>
                      <span>{comp.name}</span>

                      <Tag
                        level={level}
                        competency={comp.name}
                        specialism={specialism}
                      />
                    </div>

                    {category === 'Overview' ? (
                      <>
                        <Description
                          dangerouslySetInnerHTML={{ __html: comp.description }}
                        />
                        <Divider />
                      </>
                    ) : comp.name === 'Framework Criteria' ? (
                      <>
                        <Description
                          dangerouslySetInnerHTML={{ __html: comp.description }}
                        />
                        <Divider />
                        <div className='title'>{comp.name2}</div>
                        <p className='description'>{comp.description2}</p>
                        <Divider />
                        <div className='title'>{comp.name3}</div>
                        <p className='description'>{comp.description3}</p>
                      </>
                    ) : (
                      <>
                        <p className='description'>{comp.description}</p>
                        <Divider />
                      </>
                    )}
                  </ListItem>
                )
            )}
          </List>
        </ContentDiv>
      </ContentContainer>
    </Main>
  );
};

export default Content;
