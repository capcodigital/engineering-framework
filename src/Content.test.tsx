import * as React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Content from './Content';
import { BrowserRouter as Router } from 'react-router-dom';
import { EngineeringContextProvider } from './EngineeringContext';
import { ContentContainer, ContentHeader } from './Content';

describe('Content', () => {
  it('should render Content component', () => {
    const { container } = render(
      <Router>
        <EngineeringContextProvider>
          <Content title={'Software Engineer'} />
        </EngineeringContextProvider>
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  describe('Header', () => {
    it('should check that correct background image renders', () => {
      const { container } = render(
        <Router>
          <EngineeringContextProvider>
            <ContentHeader level={'Associate'} />
          </EngineeringContextProvider>
        </Router>
      );
      expect(container.firstChild).toHaveStyle(
        'background-image: url(associate.png)'
      );
    });

    it('should render level title with Associate Engineer text by default', () => {
      const { container } = render(
        <Router>
          <EngineeringContextProvider>
            <Content title={'Software Engineer'} />
          </EngineeringContextProvider>
        </Router>
      );
      expect(container.querySelector('.level-title')).toHaveTextContent(
        'Associate â€¢ Associate Engineer'
      );
    });

    it('should click Consultant in the side bar and render level title with Consultant Engineer text', () => {
      const { container } = render(
        <Router>
          <EngineeringContextProvider>
            <Content title={'Software Engineer'} />
          </EngineeringContextProvider>
        </Router>
      );

      fireEvent.click(screen.getByText('Consultant'));
      expect(container.querySelector('.level-title')).toHaveTextContent(
        'Consultant â€¢ Engineer'
      );
    });
  });

  it('should check that ContentContainer is hidden when menu is open on mobile', () => {
    const { container } = render(
      <Router>
        <EngineeringContextProvider>
          <ContentContainer menuOpen={true} />
        </EngineeringContextProvider>
      </Router>
    );
    expect(container.firstChild).toHaveStyle('display: none');
  });
});
