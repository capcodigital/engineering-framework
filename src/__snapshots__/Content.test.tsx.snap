// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render Content component 1`] = `
<div>
  <main
    class="Content__Main-dq2n3w-0 hXOTys"
  >
    <div
      class="SideBar__StyledSideBar-sc-16bum1j-0 gVleIT"
    >
      <div>
        <ul
          class="MuiList-root SideBarList__StyledList-sc-1egehub-0 gLBXYT list MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root level list-item active MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            Associate
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            class="MuiCollapse-container MuiCollapse-entered"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <nav
                  class="MuiList-root nested MuiList-padding"
                >
                  <span>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root category list-item active MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <a
                        href="/#overview-job-description"
                      >
                        Overview
                      </a>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                    <div
                      class="MuiCollapse-container MuiCollapse-entered"
                      style="min-height: 0px;"
                    >
                      <div
                        class="MuiCollapse-wrapper"
                      >
                        <div
                          class="MuiCollapse-wrapperInner"
                        >
                          <div
                            class="MuiList-root"
                          >
                            <div
                              aria-disabled="false"
                              class="MuiButtonBase-root MuiListItem-root nested competency list-item active  MuiListItem-gutters MuiListItem-button"
                              role="button"
                              tabindex="0"
                            >
                              <a
                                href="/#overview-job-description"
                              >
                                Job Description
                              </a>
                              <span
                                class="MuiTouchRipple-root"
                              />
                            </div>
                            <div
                              aria-disabled="false"
                              class="MuiButtonBase-root MuiListItem-root nested competency list-item   MuiListItem-gutters MuiListItem-button"
                              role="button"
                              tabindex="0"
                            >
                              <a
                                href="/#overview-you-have"
                              >
                                You Have
                              </a>
                              <span
                                class="MuiTouchRipple-root"
                              />
                            </div>
                            <div
                              aria-disabled="false"
                              class="MuiButtonBase-root MuiListItem-root nested competency list-item   MuiListItem-gutters MuiListItem-button"
                              role="button"
                              tabindex="0"
                            >
                              <a
                                href="/#overview-you-will"
                              >
                                You Will
                              </a>
                              <span
                                class="MuiTouchRipple-root"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </span>
                  <span>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root category list-item  MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <a
                        href="/#delivery-framework-criteria"
                      >
                        Delivery
                      </a>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </span>
                  <span>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root category list-item  MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <a
                        href="/#people-framework-criteria"
                      >
                        People
                      </a>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </span>
                  <span>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root category list-item  MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <a
                        href="/#stakeholder-framework-criteria"
                      >
                        Stakeholder
                      </a>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </span>
                </nav>
              </div>
            </div>
          </div>
        </ul>
        <ul
          class="MuiList-root SideBarList__StyledList-sc-1egehub-0 gLBXYT list MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root level list-item  MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            Consultant
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
        <ul
          class="MuiList-root SideBarList__StyledList-sc-1egehub-0 gLBXYT list MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root level list-item  MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            Senior Consultant
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
        <ul
          class="MuiList-root SideBarList__StyledList-sc-1egehub-0 gLBXYT list MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root level list-item  MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            Principal Consultant
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
        <ul
          class="MuiList-root SideBarList__StyledList-sc-1egehub-0 gLBXYT list MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root level list-item  MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            Managing Principal
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
    </div>
    <div
      class="BreadcrumbsNav__Wrapper-sc-146cviq-0 gbeLud"
    >
      <div
        class="breadcrumbs-wrapper"
      >
        <div
          class="title"
        >
          Digital Engineering
        </div>
        <nav
          aria-label="breadcrumb"
          class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          <ol
            class="MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <div>
                Associate
              </div>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <div>
                Overview
              </div>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <div>
                Job Description
              </div>
            </li>
          </ol>
        </nav>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
    <div
      class="Content__ContentContainer-dq2n3w-1 eqntqX"
    >
      <div
        class="Content__ContentHeader-dq2n3w-2 nNpiq"
      >
        <div
          class="level-title"
        >
          <span
            class="level"
          >
            Associate
          </span>
           •
           
          Associate Engineer
        </div>
        <div
          class="category"
        >
          Overview
        </div>
      </div>
      <div
        class="Content__ContentDiv-dq2n3w-3 lhIFjD"
      >
        <div
          class="MuiList-root competency-list MuiList-padding"
        >
          <li
            class="MuiListItem-root list-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
            id="overview-job-description"
          >
            <div
              class="title"
            >
              <span>
                Job Description
              </span>
            </div>
            <p
              class="Content__Description-dq2n3w-4 jMEZjF"
            >
              An associate developer has limited experience and is relatively new to the practice of software development.  Their focus should be on acquiring skills and improving their capabilities.  On the consulting side they should be focused on improving their communication and team work skills as these will be important for future leadership opportunities.
            </p>
            <hr
              class="MuiDivider-root"
            />
          </li>
          <li
            class="MuiListItem-root list-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
            id="overview-you-have"
          >
            <div
              class="title"
            >
              <span>
                You Have
              </span>
            </div>
            <p
              class="Content__Description-dq2n3w-4 jMEZjF"
            >
              <ul>
                <li>
                  Theoretical background such as a university or college program in Computer Science (Technology)
                </li>
                <li>
                  Coop / internship or equivalent work experience
                </li>
                <li>
                  Desire / drive to scale your business and technology knowledge
                </li>
              </ul>
            </p>
            <hr
              class="MuiDivider-root"
            />
          </li>
          <li
            class="MuiListItem-root list-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
            id="overview-you-will"
          >
            <div
              class="title"
            >
              <span>
                You Will
              </span>
            </div>
            <p
              class="Content__Description-dq2n3w-4 jMEZjF"
            >
              <ul>
                <li>
                  Focus on increasing competencies in Software Development
                </li>
                <li>
                  Enhance communication and relationship management skills with stakeholders.
                </li>
              </ul>
            </p>
            <hr
              class="MuiDivider-root"
            />
          </li>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`should render Content component 1`] = `
<div>
  <main
    class="Content__Main-dq2n3w-0 hXOTys"
  >
    <div
      class="SideBar__StyledSideBar-sc-16bum1j-0 gVleIT"
    >
      <div>
        <ul
          class="MuiList-root SideBarList__StyledList-sc-1egehub-0 gLBXYT list MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root level list-item active MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            Associate
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            class="MuiCollapse-container MuiCollapse-entered"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <nav
                  class="MuiList-root nested MuiList-padding"
                >
                  <span>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root category list-item active MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <a
                        href="/#overview-job-description"
                      >
                        Overview
                      </a>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                    <div
                      class="MuiCollapse-container MuiCollapse-entered"
                      style="min-height: 0px;"
                    >
                      <div
                        class="MuiCollapse-wrapper"
                      >
                        <div
                          class="MuiCollapse-wrapperInner"
                        >
                          <div
                            class="MuiList-root"
                          >
                            <div
                              aria-disabled="false"
                              class="MuiButtonBase-root MuiListItem-root nested competency list-item active  MuiListItem-gutters MuiListItem-button"
                              role="button"
                              tabindex="0"
                            >
                              <a
                                href="/#overview-job-description"
                              >
                                Job Description
                              </a>
                              <span
                                class="MuiTouchRipple-root"
                              />
                            </div>
                            <div
                              aria-disabled="false"
                              class="MuiButtonBase-root MuiListItem-root nested competency list-item   MuiListItem-gutters MuiListItem-button"
                              role="button"
                              tabindex="0"
                            >
                              <a
                                href="/#overview-you-have"
                              >
                                You Have
                              </a>
                              <span
                                class="MuiTouchRipple-root"
                              />
                            </div>
                            <div
                              aria-disabled="false"
                              class="MuiButtonBase-root MuiListItem-root nested competency list-item   MuiListItem-gutters MuiListItem-button"
                              role="button"
                              tabindex="0"
                            >
                              <a
                                href="/#overview-you-will"
                              >
                                You Will
                              </a>
                              <span
                                class="MuiTouchRipple-root"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </span>
                  <span>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root category list-item  MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <a
                        href="/#delivery-framework-criteria"
                      >
                        Delivery
                      </a>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </span>
                  <span>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root category list-item  MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <a
                        href="/#people-framework-criteria"
                      >
                        People
                      </a>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </span>
                  <span>
                    <div
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiListItem-root category list-item  MuiListItem-gutters MuiListItem-button"
                      role="button"
                      tabindex="0"
                    >
                      <a
                        href="/#stakeholder-framework-criteria"
                      >
                        Stakeholder
                      </a>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </span>
                </nav>
              </div>
            </div>
          </div>
        </ul>
        <ul
          class="MuiList-root SideBarList__StyledList-sc-1egehub-0 gLBXYT list MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root level list-item  MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            Consultant
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
        <ul
          class="MuiList-root SideBarList__StyledList-sc-1egehub-0 gLBXYT list MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root level list-item  MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            Senior Consultant
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
        <ul
          class="MuiList-root SideBarList__StyledList-sc-1egehub-0 gLBXYT list MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root level list-item  MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            Principal Consultant
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
        <ul
          class="MuiList-root SideBarList__StyledList-sc-1egehub-0 gLBXYT list MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root level list-item  MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            Managing Principal
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
    </div>
    <div
      class="BreadcrumbsNav__Wrapper-sc-146cviq-0 gbeLud"
    >
      <div
        class="breadcrumbs-wrapper"
      >
        <div
          class="title"
        >
          Digital Engineering
        </div>
        <nav
          aria-label="breadcrumb"
          class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          <ol
            class="MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <div>
                Associate
              </div>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <div>
                Overview
              </div>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <div>
                Job Description
              </div>
            </li>
          </ol>
        </nav>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
    <div
      class="Content__ContentContainer-dq2n3w-1 eqntqX"
    >
      <div
        class="Content__ContentHeader-dq2n3w-2 nNpiq"
      >
        <div
          class="level-title"
        >
          <span
            class="level"
          >
            Associate
          </span>
           •
           
          Associate Engineer
        </div>
        <div
          class="category"
        >
          Overview
        </div>
      </div>
      <div
        class="Content__ContentDiv-dq2n3w-3 lhIFjD"
      >
        <div
          class="MuiList-root competency-list MuiList-padding"
        >
          <li
            class="MuiListItem-root list-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
            id="overview-job-description"
          >
            <div
              class="title"
            >
              <span>
                Job Description
              </span>
            </div>
            <p
              class="Content__Description-dq2n3w-4 jMEZjF"
            >
              An associate developer has limited experience and is relatively new to the practice of software development.  Their focus should be on acquiring skills and improving their capabilities.  On the consulting side they should be focused on improving their communication and team work skills as these will be important for future leadership opportunities.
            </p>
            <hr
              class="MuiDivider-root"
            />
          </li>
          <li
            class="MuiListItem-root list-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
            id="overview-you-have"
          >
            <div
              class="title"
            >
              <span>
                You Have
              </span>
            </div>
            <p
              class="Content__Description-dq2n3w-4 jMEZjF"
            >
              <ul>
                <li>
                  Theoretical background such as a university or college program in Computer Science (Technology)
                </li>
                <li>
                  Coop / internship or equivalent work experience
                </li>
                <li>
                  Desire / drive to scale your business and technology knowledge
                </li>
              </ul>
            </p>
            <hr
              class="MuiDivider-root"
            />
          </li>
          <li
            class="MuiListItem-root list-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
            id="overview-you-will"
          >
            <div
              class="title"
            >
              <span>
                You Will
              </span>
            </div>
            <p
              class="Content__Description-dq2n3w-4 jMEZjF"
            >
              <ul>
                <li>
                  Focus on increasing competencies in Software Development
                </li>
                <li>
                  Enhance communication and relationship management skills with stakeholders.
                </li>
              </ul>
            </p>
            <hr
              class="MuiDivider-root"
            />
          </li>
        </div>
      </div>
    </div>
  </main>
</div>
`;
